pkg/kudoctl/packages/reader/read_dir.go

```diff
@@ -26,13 +27,13 @@ func ReadDir(fs afero.Fs, path string) (*packages.Package, error) {
 	// 1. get files
 	files, err := FromDir(fs, path)
 	if err != nil {
-		return nil, errors.Wrap(err, "while parsing package files")
+		return nil, fmt.Errorf("while parsing package files: %w", err)
```

Comment:
I don't understand why are we wrapping the error here, do we need in the upper layers distinguish between some error type? Otherwise this would end up in a pretty weird error message printed out.

btw. I would say Wrapf is more %v than %w. I think %w has a bit of different use cases. Recommend reading https://blog.golang.org/go1.13-errors including when to wrap (in case you did not so far)

Reply:
Well, with regards to %v vs %w: From what I understand (and see from testing) %w just wraps the error, and allows it to be unwrappable - apart from that it behaves like %v. Which says to me, it's generally a good idea to use %w instead of %v.

```
In Go 1.13, the fmt.Errorf function supports a new %w verb. When this verb is present, the error returned by fmt.Errorf will have an Unwrap method returning the argument of %w, which must be an error. In all other ways, %w is identical to %v.
```

Reply:
I just have this question in another PR. To quote that same blog post:

```
In other words, wrapping an error makes that error part of your API. If you don't want to 
commit to supporting that error as part of your API in the future, you shouldn't wrap the error.

It’s important to remember that whether you wrap or not, the error text will be the same. A 
person trying to understand the error will have the same information either way; the choice to 
wrap is about whether to give programs additional information so they can make more informed 
decisions, or to withhold that information to preserve an abstraction layer.
```

We should get an agreement on this. Personally, I'm leaning towards:
- Using `%v` by default to preserve abstractions boundaries
- Using `%w` intentionally when exposing the underlying error type is necessary

Thoughts?

Reply:
+1 to what @zen-dog wrote

Reply:
I do tend to lean towards: More information is better than less information, but I do get the abstraction boundaries. Coming from Java, I hate to not wrap exceptions, because of the enclosed stack traces, but as we don't have them here, I should probably adjust. 

So, I can agree with @zen-dog proposal to use `%v` per default any only using `%w` when we expect to unwrap on the caller side. I don't think there needs to be an existing unwrap, but a high possibility and use case for an unwrap to happen though.

Reply:
I've changed the `%w`s to `%v`. I don't think in this area we don't need to wrap any errors